src/benchmarks/micro/corefx/System.Runtime/Perf.Enum.cs

```diff
@@ -20,19 +22,534 @@ public enum Colors
+        public string ToStringFormat(Enum value, string format) => value.ToString(format);
 
         [Benchmark]
-        public bool HasFlag() => _greenAndRed.HasFlag(Colors.Green);
+        public string ToStringFormatF() => (Colors.Green | Colors.Yellow | Colors.Orange).ToString("F");
+        #endregion
+
+        #region Format
+        [Benchmark]
+        [Arguments(typeof(BigContiguousEnum), BigContiguousEnum.C, "D")]
+        [Arguments(typeof(BigContiguousEnum), BigContiguousEnum.C, "X")]
+        [Arguments(typeof(BigContiguousEnum), BigContiguousEnum.C, "G")]
+        //[Arguments(typeof(Colors), Colors.Green | Colors.Yellow | Colors.Orange, "F")] // BenchmarkDotNet Issue #1020
+        public string Format(Type enumType, Enum value, string format) => Enum.Format(enumType, value, format);
+
+        [Benchmark]
+        [Arguments(typeof(Colors))]
+        public string FormatF(Type enumType) => Enum.Format(enumType, Colors.Green | Colors.Yellow | Colors.Orange, "F");
+        #endregion
+
+        [Benchmark]
+        [Arguments(BigContiguousEnum.C, BigContiguousEnum.E)]
+        public int CompareToEnum(Enum value, object target) => value.CompareTo(target);
+
+        [Benchmark]
+        [Arguments(BigContiguousEnum.C)]
+        public int CompareToNull(Enum value) => value.CompareTo(null);
+
+        [Benchmark]
+        [Arguments(Int32Enum.One)]
+        public TypeCode GetTypeCode(Enum value) => value.GetTypeCode();
+
+        #region IConvertible
+        [Benchmark]
+        [Arguments(ByteEnum.Max)]
+        public byte ToByte(IConvertible value) => value.ToByte(null);
+
+        [Benchmark]
+        [Arguments(SByteEnum.Max)]
+        public sbyte ToSByte(IConvertible value) => value.ToSByte(null);
+
+        [Benchmark]
+        [Arguments(Int16Enum.Max)]
+        public short ToInt16(IConvertible value) => value.ToInt16(null);
+
+        [Benchmark]
+        [Arguments(UInt16Enum.Max)]
+        public ushort ToUInt16(IConvertible value) => value.ToUInt16(null);
+
+        [Benchmark]
+        [Arguments(Int32Enum.Max)]
+        public int ToInt32(IConvertible value) => value.ToInt32(null);
+
+        [Benchmark]
+        [Arguments(UInt32Enum.Max)]
+        public uint ToUInt32(IConvertible value) => value.ToUInt32(null);
+
+        [Benchmark]
+        [Arguments(Int64Enum.Max)]
+        public long ToInt64(IConvertible value) => value.ToInt64(null);
+
+        [Benchmark]
+        [Arguments(UInt64Enum.Max)]
+        public ulong ToUInt64(IConvertible value) => value.ToUInt64(null);
+        #endregion
+
+        #region Parsing
+        [Benchmark]
+        [Arguments(typeof(Colors))]
+        public object ParseFlagsNonGeneric(Type enumType) => Enum.Parse(enumType, "Red, Orange, Yellow, Green, Blue");
+
+        [Benchmark]
+        public Colors ParseFlagsGeneric() => Enum.Parse<Colors>("Red, Orange, Yellow, Green, Blue");
+
+        [Benchmark]
+        [Arguments(typeof(ByteEnum), "Two")]
```

Comment:
could you please use "nameof" here and in the examples below?
```suggestion
        [Arguments(typeof(ByteEnum), nameof(ByteEnum.Two))]
```